strawberryfields/engine.py

```diff
@@ -336,8 +360,144 @@ def run(self, program, *, shots=1, compile_options={}, modes=None, state_options
             # empty sequence
             pass
         else:
-            result.state = self.backend.state(modes, **state_options)  # tfbackend.state can use kwargs
+            result.state = self.backend.state(
+                modes, **state_options
+            )  # tfbackend.state can use kwargs
         return result
 
 
+class StarshipEngine(BaseEngine):
+    """
+    Starship quantum program executor engine.
+
+    Executes :class:`.Program` instances on the chosen remote backend, and makes
+    the results available via :class:`.Result`.
+
+    Args:
+        backend (str, BaseBackend): name of the backend, or a pre-constructed backend instance
+    """
+
+    API_DEFAULT_REFRESH_SECONDS = 1
+    HARDWARE_BACKENDS = ("chip0",)
```

Comment:
I think this is fine for now, but for future PRs, I'm a bit wary about hardcoding available hardware backends directly into the StarshipEngine, as the hardware availability might fluctuate independently of SF releases.

Once the corresponding API endpoints are available, we should add support for

* `StrarshipEngine.backend_info()`: show a list of available devices/simulators/backends

* `StarshipEngine.backend_info('chip0')`: shows more detailed information and stats about a specific backend

and use the results to provide the logic for which backends are available.

Reply:
This makes sense, and will make this more modular. I will keep a note of this.